File: SDKs/googleapis.com/deploymentmanager/v2/go/pkg/models/shared/manifest.go
package shared

type Manifest struct {
	Config                 *ConfigFile  `json:"config"`
	ExpandedConfig         *string      `json:"expandedConfig"`
	ID                     *string      `json:"id"`
	Imports                []ImportFile `json:"imports"`
	InsertTime             *string      `json:"insertTime"`
	Layout                 *string      `json:"layout"`
	ManifestSizeBytes      *string      `json:"manifestSizeBytes"`
	ManifestSizeLimitBytes *string      `json:"manifestSizeLimitBytes"`
	Name                   *string      `json:"name"`
	SelfLink               *string      `json:"selfLink"`
}

File: SDKs/googleapis.com/deploymentmanager/v2beta/typescript/src/sdk/models/shared/manifest.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { ConfigFile } from "./configfile";
import { ImportFile } from "./importfile";



export class Manifest extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=config" })
  config?: ConfigFile;

  @SpeakeasyMetadata({ data: "json, name=expandedConfig" })
  expandedConfig?: string;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=imports", elemType: ImportFile })
  imports?: ImportFile[];

  @SpeakeasyMetadata({ data: "json, name=insertTime" })
  insertTime?: string;

  @SpeakeasyMetadata({ data: "json, name=layout" })
  layout?: string;

  @SpeakeasyMetadata({ data: "json, name=manifestSizeBytes" })
  manifestSizeBytes?: string;

  @SpeakeasyMetadata({ data: "json, name=manifestSizeLimitBytes" })
  manifestSizeLimitBytes?: string;

  @SpeakeasyMetadata({ data: "json, name=name" })
  name?: string;

  @SpeakeasyMetadata({ data: "json, name=selfLink" })
  selfLink?: string;
}
